External/Plugins/ASClassWizard/PluginMain.cs

```diff
@@ -227,28 +227,37 @@ private void DisplayClassWizard(String inDirectory, String templateFile, String
-            bool isHaxe = project.Language == "haxe";
-            dialog.Project = project;
-            dialog.Directory = inDirectory;
-			dialog.StartupClassName = className;
-            if (package != null)
+            using (AS3ClassWizard dialog = new AS3ClassWizard())
             {
-                package = package.Replace(Path.DirectorySeparatorChar, '.');
-                dialog.StartupPackage = package;
-            }
-            if (dialog.ShowDialog() == DialogResult.OK)
-            {
-                string cPackage = dialog.getPackage();
-                string path = Path.Combine(classpath, dialog.getPackage().Replace('.', Path.DirectorySeparatorChar));
-                string newFilePath  = Path.ChangeExtension(Path.Combine(path, dialog.getClassName()), isHaxe ? ".hx" : ".as");
-                if (File.Exists(newFilePath))
+                bool isHaxe = project.Language == "haxe";
+                dialog.Project = project;
+                dialog.Directory = inDirectory;
+                dialog.StartupClassName = className;
+                if (package != null)
                 {
-                    string title = " " + TextHelper.GetString("FlashDevelop.Title.ConfirmDialog");
-                    string message = TextHelper.GetString("PluginCore.Info.FolderAlreadyContainsFile");
-                    DialogResult result = MessageBox.Show(PluginBase.MainForm, string.Format(message, newFilePath, "\n"), title, MessageBoxButtons.YesNoCancel, MessageBoxIcon.Warning);
-                    if (result == DialogResult.Cancel) return;
+                    package = package.Replace(Path.DirectorySeparatorChar, '.');
+                    dialog.StartupPackage = package;
                 }
+                DialogResult conflictResult = DialogResult.OK;
+                string cPackage, path, newFilePath;
+                do
+                {
+                    if (dialog.ShowDialog() != DialogResult.OK) return;
+                    cPackage = dialog.getPackage();
+                    path = Path.Combine(classpath, cPackage.Replace('.', Path.DirectorySeparatorChar));
+                    newFilePath = Path.ChangeExtension(Path.Combine(path, dialog.getClassName()),
+                                                              isHaxe ? ".hx" : ".as");
+                    if (File.Exists(newFilePath))
+                    {
+                        string title = " " + TextHelper.GetString("FlashDevelop.Title.ConfirmDialog");
+                        string message = TextHelper.GetString("PluginCore.Info.FolderAlreadyContainsFile");
+                        conflictResult = MessageBox.Show(PluginBase.MainForm,
+                            string.Format(message, newFilePath, "\n"), title,
+                            MessageBoxButtons.YesNoCancel, MessageBoxIcon.Warning);
+                        if (conflictResult == DialogResult.No) return;
+                    }
+                } while (conflictResult == DialogResult.Cancel);
```

Comment:
Before, the logic was:

Show wizard dialog.
If the dialog is accepted:
  - Get the new file name.
  - Check for conflict, if conflict:
       - Display a YesNoCancel alert, if the user pressed Cancel the function returned, otherwise the file was overwritten.

Since presenting a YesNoCancel message box is standard, and the cancel option shouldn't cause any data loss, selecting it displays the dialog again with all fields filled. I think the old option of selecting a YesNoCancel dialog was quite appropiate, just Yes and No would made the user to launch and fill the form again.

I wouldn't say it's doing anything fancy, it's just wrapping the minimal code needed in a do while loop, and also checking for the No option that was before ignored, the rest of the code is the same as before.